src/logic/calculate.js

```diff
@@ -1,58 +1,141 @@
 import operate from './operate';
 
-const calculate = (data, buttonName) => {
-  let { total, next, operation } = data;
-  const numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
-  const operators = ['+', '-', '*', '/', '%', '='];
-
-  if (numbers.includes(buttonName)) {
-    if (total === '0' && !total) {
-      total = buttonName;
-    }
-    if (total && !operation && !next && total !== 0) {
-      total += buttonName;
-    }
-    if (total && operation && !next) {
-      next = buttonName;
-    }
-    if (total && operation && next) {
-      next += buttonName;
-    }
-  }
+// const calculate = (data, buttonName) => {
+//   let { total, next, operation } = data;
+//   const numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+//   const operators = ['+', '-', '*', '/', '%', '='];
 
-  if (buttonName === '+/-') {
-    if (total && !next) {
-      total = operate(total, -1, '*');
-    } else if (total && next) {
-      next = operate(next, -1, '*');
-    }
-  }
+//   if (numbers.includes(buttonName)) {
+//     if (total === '0' && !total) {
+//       total = buttonName;
+//     }
+//     if (total && !operation && !next && total !== 0) {
+//       total += buttonName;
+//     }
+//     if (total && operation && !next) {
+//       next = buttonName;
+//     }
+//     if (total && operation && next) {
+//       next += buttonName;
+//     }
+//   }
 
-  if (buttonName === '.') {
-    if (!total && !next) {
-      total = '0.';
-    } else if (total && !next && total.indexOf('.') === -1) {
-      total += '.';
-    } else if (next && next.indexOf('.')) {
-      next += '.';
-    }
-  }
+//   if (buttonName === '+/-') {
+//     if (total && !next) {
+//       total = operate(total, -1, '*');
+//     } else if (total && next) {
+//       next = operate(next, -1, '*');
+//     }
+//   }
 
-  if (buttonName === 'AC') {
-    total = 0;
-    next = '';
-    operation = '';
-  }
+//   if (buttonName === '.') {
+//     if (!total && !next) {
+//       total = '0.';
+//     } else if (total && !next && total.indexOf('.') === -1) {
+//       total += '.';
+//     } else if (next && next.indexOf('.')) {
+//       next += '.';
+//     }
+//   }
+
+//   if (buttonName === 'AC') {
+//     total = 0;
+//     next = '';
+//     operation = '';
+//   }
+
+//   if (operators.includes(buttonName)) {
+//     if (total && !next) {
+//       operation = buttonName;
+//     } else if (total && operation && next) {
+//       total = operate(total, next, operation);
+//     }
+//   }
+
+//   return total;
+// };
 
-  if (operators.includes(buttonName)) {
-    if (total && !next) {
-      operation = buttonName;
-    } else if (total && operation && next) {
+// const calculate = (dataObj, btnName) =>{
+//   let {total, next, operation} = dataObj;
+//   const check =()=>{
+//     if(next && operation){
+//       total = operate(total, next,ope)
+//     }
+//   }
+
+// }
+
```

Comment:
### Optional
- [ ] It would be better if we remove the commented piece of code

Reply:
All comments are deleted